modules/context/repo.go

```diff
@@ -319,7 +319,14 @@ func EarlyResponseForGoGetMeta(ctx *Context) {
 		ctx.PlainText(http.StatusBadRequest, "invalid repository path")
 		return
 	}
-	goImportContent := fmt.Sprintf("%s git %s", ComposeGoGetImport(username, reponame), repo_model.ComposeHTTPSCloneURL(username, reponame))
+
+	var cloneURL string
+	if setting.Repository.UseGitCloneURL {
```

Comment:
Are there any go get parameter to indicate a ssh or a http protocol?

Reply:
yes

Reply:
I haven't looked into details nor used "go get over ssh", so I'm not sure whether my understanding is correct, according to:

1. the previous comment https://github.com/go-gitea/gitea/issues/12192#issuecomment-657190979
2. the fact that how `go get` works: it only reads the `github.com/owner/repo` in `go.mod`: there is no chance for `go get` to pass some "ssh/https" parameter.

So, I would assume this PR's approach is right. Correct me if I am wrong.

Reply:
Even if we need a new configuration, the name `UseGitCloneURL` is still confusing. Maybe `GoGetProtocol` is a better name?